src/alerts/check_signals_updates.R

```diff
@@ -0,0 +1,207 @@
+box::use(jsonlite[fromJSON, toJSON])
+box::use(httr[GET, POST, add_headers, status_code, content])
+box::use(purrr)
+box::use(logger[log_error, log_info])
+
+box::use(../utils/get_env[get_env])
+box::use(../utils/hs_logger)
+box::use(cs = ../utils/cloud_storage)
+
+hs_logger$configure_logger()
+
+
+#' Queries the GitHub API to get all runs from a given monitoring workflow
+#' Requires workflows to be named following this convention: `monitor_<indicator_id>.yaml`
+#'
+#' NOTE: More data will be returned as each workflow is run more times. Perhaps won't scale well,
+#' depending on how many runs a workflow will have
+#'
+#' @param indicator_id ID of the indicator
+#'
+#' @returns JSON API response
+gh_status <- function(indicator_id) {
+  # TODO Temp cleaning before workflows are renamed to match indicator ids
+  ind <- sub("^[^_]*_", "", indicator_id)
+  workflow_id <- paste0("monitor_", ind, ".yaml")
+  token <- get_env("GH_TOKEN")
+  url <- paste0("https://api.github.com/repos/ocha-dap/hdx-signals/actions/workflows/", workflow_id, "/runs")
+  response <- GET(url, add_headers(Authorization = paste("token", token)))
+  return(response)
+}
+
+#' Builds and posts a message to Slack, using incoming webhooks
+#' See API docs: https://api.slack.com/messaging/webhooks
+#'
+#' @param header_text Text for the message header
+#' @param status_text Text for the GitHub actions status report
+#' @param signals_text Text for the signals status report
+#'
+#' @returns Nothing. Message is posted to slack and will log an error if not successful
+slack_post_message <- function(header_text, status_text, signals_text) {
+  # See https://app.slack.com/block-kit-builder for prototyping layouts in JSON
+  msg <- list(
+    blocks = list(
+      list(
+        type = "section",
+        text = list(
+          type = "plain_text",
+          text = header_text,
+          emoji = TRUE
+        )
+      ),
+      list(
+        type = "section",
+        text = list(
+          type = "mrkdwn",
+          text = signals_text
+        )
+      ),
+      list(type = "divider"),
+      list(
+        type = "context",
+        elements = list(
+          list(
+            type = "mrkdwn",
+            text = status_text
+          )
+        )
+      ),
+      list(type = "divider")
+    )
+  )
+  json_body <- toJSON(msg, pretty = TRUE, auto_unbox = TRUE)
+  response <- POST(get_env("SLACK_URL"), body = json_body, encode = "json")
+  if (response$status != 200) {
+    log_error("Error posting Slack message")
```

Comment:
With `{httr2}`, you can capture errors by adding `httr2$req_error()` in the pipe above, and wrapping there.